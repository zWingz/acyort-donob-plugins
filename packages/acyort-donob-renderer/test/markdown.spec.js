const { join } = require('path')
const fs = require('fs-extra')
const parseMd = require('../lib/markdown')

describe('test markdown parse', () => {
  it('test heading parse', () => {
    const content = fs.readFileSync(join(__dirname, './fixtures/parse.md'), {
      encoding: 'utf8',
    })
    expect(parseMd(content)).toMatchSnapshot('parseMd')
  })
  it('test parse code', () => {
    const code = `
\`\`\`javascript
const a = 1
const b = 1
const c = a + b
\`\`\`
    `
    const content = parseMd(code)
    expect(content).toMatchSnapshot('parseCode')
  })
  it('test code with lineNumber', () => {
    const code = `
\`\`\`js
const a = 1
const b = 1
const c = a + b
\`\`\`
`
    const content = parseMd(code, {
      lineNumber: true,
    })
    expect(content).toMatchSnapshot('parseCodeLineNumber')
  })
  it('test code with uknow lang', () => {
    const c = `
\`\`\`fdsa
const a = 1
const b = 1
const c = a + b
\`\`\`
    `
    const content = parseMd(c)
    expect(content).toMatchSnapshot('parseCodeUnknowLang')
  })
})
