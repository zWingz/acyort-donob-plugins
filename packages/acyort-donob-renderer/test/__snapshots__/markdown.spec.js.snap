// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test markdown parse test code with lineNumber: parseCodeLineNumber 1`] = `
"
      <div class=\\"hljs javascript\\">
        <table>
          <tbody>
            <tr>
              <td class=\\"line\\">
                <pre><span>1</span>
<span>2</span>
<span>3</span>
</pre>
              </td>
              <td class=\\"code\\"><pre>const <span class=\\"hljs-selector-tag\\">a</span> = <span class=\\"hljs-number\\">1</span>
const <span class=\\"hljs-selector-tag\\">b</span> = <span class=\\"hljs-number\\">1</span>
const c = <span class=\\"hljs-selector-tag\\">a</span> + b</pre></td>
            </tr>
          </tbody>
        </table>
      </div>
    "
`;

exports[`test markdown parse test code with uknow lang: parseCodeUnknowLang 1`] = `
"
      <div class=\\"hljs code\\">
        <pre>const a &#x3D; 1
const b &#x3D; 1
const c &#x3D; a + b</pre>
      </div>
    "
`;

exports[`test markdown parse test heading parse: parseMd 1`] = `
"
    <h1 class=\\"heading\\">
      <a href=\\"#marked-in-the-browsernnrendered-by-marked\\" id=\\"marked-in-the-browsernnrendered-by-marked\\" class=\\"heading-anchor\\"></a>Marked in the browser\\\\n\\\\nRendered by <strong>marked</strong>.
    </h1>
  
    <h2 class=\\"heading\\">
      <a href=\\"#section-one\\" id=\\"section-one\\" class=\\"heading-anchor\\"></a>Section One
    </h2>
  
    <h2 class=\\"heading\\">
      <a href=\\"#hello\\" id=\\"hello\\" class=\\"heading-anchor\\"></a><a href=\\"hello\\">hello</a>
    </h2>
  
    <h3 class=\\"heading\\">
      <a href=\\"#test\\" id=\\"test\\" class=\\"heading-anchor\\"></a><code>test</code>
    </h3>
  
    <h2 class=\\"heading\\">
      <a href=\\"#%E5%95%8A%E5%95%8A%E5%95%8A\\" id=\\"%E5%95%8A%E5%95%8A%E5%95%8A\\" class=\\"heading-anchor\\"></a>啊啊啊
    </h2>
  "
`;

exports[`test markdown parse test parse code: parseCode 1`] = `
"
      <div class=\\"hljs javascript code\\">
        <pre>const <span class=\\"hljs-selector-tag\\">a</span> = <span class=\\"hljs-number\\">1</span>
const <span class=\\"hljs-selector-tag\\">b</span> = <span class=\\"hljs-number\\">1</span>
const c = <span class=\\"hljs-selector-tag\\">a</span> + b</pre>
      </div>
    "
`;
